code/rust-sokoban-c03-05/Cargo.toml

```diff
@@ -6,6 +6,8 @@ edition = "2018"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+cargo-features = ["unstable"]
```

Comment:
can we can avoid using unstable? trying to keep the tutorial accessible to beginners and I worry that if we start going down the unstable path more unstable code would make its way in. I do agree that drain filter is the nicest most idiomatic API for this, but I would be happy with some clones/etc if it made it easier to explain to new folks. we can always put a little note in the text saying the "best" way to do this is unstable+drain, check it out here.

Reply:
Yeah, I'm not sure what the cleanest implementation of it would look like. Like I said, this is just what I have done after not enjoying the large match in my project using the event_queue pattern. 

And to be fair, it may be easier to read for some if it just collected filtered vec into a variable and then for loop iterate over that.